crates/sui-source-validation/src/tests.rs

```diff
@@ -252,26 +258,32 @@ async fn package_not_found() -> anyhow::Result<()> {
     let client = context.get_client().await?;
     let verifier = BytecodeSourceVerifier::new(client.read_api(), false);
 
-    assert!(matches!(
-        verifier.verify_package_deps(&a_pkg.package).await,
-        Err(SourceVerificationError::SuiObjectRefFailure(_)),
-    ),);
+    let Err(err) = verifier.verify_package_deps(&a_pkg.package).await else {
+        panic!("Expected verification to fail");
+    };
 
-    assert!(matches!(
-        // Subst address here doesnt matter
-        verifier
-            .verify_package_root_and_deps(&a_pkg.package, AccountAddress::random())
-            .await,
-        Err(SourceVerificationError::SuiObjectRefFailure(_)),
-    ),);
+    let expected = expect!["Dependency object does not exist or was deleted: ObjectNotFound { object_id: <id>, version: None }"];
+    expected.assert_eq(&sanitize_id(&err.to_string()));
```

Comment:
The one thing I did run into is that object IDs are randomly generated, and I couldn't find a way to configure these to be deterministic for tests (e.g., in publish_package). My idea is to sanitize the value to `<id>` with a regex helper, this is a bit janky, but probably OK for a test. LMK if there's something better available here.

Reply:
Yeah, this is an issue/annoying.  I suppose `expected` can't accept custom/format strings (It would make sense, how would it update them automatically if it could).

We have a test mode (`simtest`) that makes these fixed between runs, assuming the code doesn't change, but the IDs can change for unexpected reasons (some part of their seed changing) which leads to noisy diffs, so this is probably the best path forward.